llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -227,15 +231,26 @@ static const unsigned MaxPHINumOperands = 128;
 /// avoids spending time checking the cost model and realizing that they will
 /// be inevitably scalarized.
 static bool isValidElementType(Type *Ty) {
+  if (RunSLPReVectorization && isa<FixedVectorType>(Ty))
+    Ty = Ty->getScalarType();
```

Comment:
Do you need the check here? Better just use Ty->getScalarTy() directly here and check for RunSLPReVectorization only where the vectorization is requested

Reply:
We should keep `SLPReVec` since some FixedVectorType are accidentally passed into `isValidElementType`.